src-graphql/client.js

```diff
@@ -7,6 +7,29 @@ import productConnectionQuery from './product-connection-query';
 import collectionQuery from './collection-query';
 import collectionConnectionQuery from './collection-connection-query';
 
+function fetchAll(paginatedModels, client) {
+  return client.fetchNextPage(paginatedModels).then(({model}) => {
+    // Until we know how hasNextPage will be exposed, we query until the result is empty
+    if (model.length === 0) {
+      return paginatedModels;
+    }
+
+    paginatedModels.push(...model);
+
+    return fetchAll(paginatedModels, client);
+  });
+}
+
+function saturateProducts(promises, productData, product, client) {
```

Comment:
I don't mind it... but it's literally just doing some `fetchAll` calls. Something more literal would be `fetchAllProductResources` or something. If we aren't mutating the promises in here then `saturate` makes less sense I think too?